libr/core/anal_objc.c

```diff
@@ -194,6 +194,10 @@ static bool objc_find_refs(RCore *core) {
 		}
 
 		int count = readDword (&objc, classMethodsVA + 4);
+		if (count == -1) {
```

Comment:
i think it will be better to make count ut32 (instead of int) and check for UT32_MAX instead of -1

Reply:
and probably we can `break` instead of `continue` because such read may result in longer loading times

Reply:
Also i have observed the return value of `r_io_read_at` inside the `readDword` function is ignored, it will be good to check for that and return UT32_MAX in case of failure. As long as ut32-max is a valid ut32 value maybe worth to return an error as 3rd argument as reference

Reply:
So correct me if I am wrong but it is possible to get an invalid count before the end of the analysis, but getting a valid count afterwards, doesn't it?
If we `break` there it will stop the analysis all at once possibly missing the rest of the file.

As for the other comments, fixing now :)

Reply:
it's not the same reading 0xffffffff than being unable to read from a specific address, so the errors shouldnt be combined with the data logic.

Thanks for the fixes!

Reply:
Oh I get what you mean.
Fixed.